src/controller/event/handler/replication/replication.go

```diff
@@ -57,13 +60,16 @@ func (r *Handler) IsStateful() bool {
 func (r *Handler) handlePushArtifact(event *event.PushArtifactEvent) error {
 	art := event.Artifact
 	public := false
-	project, err := project.Mgr.Get(art.ProjectID)
-	if err == nil && project != nil {
-		public = project.IsPublic()
-	} else {
-		log.Error(err)
+	project, err := project2.Ctl.Get(context.TODO(), art.ProjectID, project2.Metadata(true))
+	if err != nil {
+		log.Errorf("failed to get project: %d, error: %v", art.ProjectID, err)
+		return err
+	}
+	if project == nil {
+		return fmt.Errorf("project %d not found of replication event", art.ProjectID)
 	}
```

Comment:
This if statement is not needed because the `Get` method of the project controller will return `not found` error when the project not exists.

```go
func (c *controller) Get(ctx context.Context, projectID int64, options ...Option) (*models.Project, error) {
	p, err := c.projectMgr.Get(projectID)
	if err != nil {
		return nil, err
	}
	if p == nil {
		return nil, errors.NotFoundError(nil).WithMessage("project %d not found", projectID)
	}

	return c.assembleProject(ctx, p, newOptions(options...))
}
```

Reply:
done